source/modulo_components/test/cpp/test_component_interface.cpp

```diff
@@ -41,6 +41,44 @@ class ComponentInterfaceTest : public ::testing::Test {
 using NodeTypes = ::testing::Types<rclcpp::Node, rclcpp_lifecycle::LifecycleNode>;
 TYPED_TEST_SUITE(ComponentInterfaceTest, NodeTypes);
 
+TYPED_TEST(ComponentInterfaceTest, RatePeriodParameters) {
+  std::shared_ptr<ComponentInterfacePublicInterface<TypeParam>> component;
+  auto node_options = rclcpp::NodeOptions();
+  if (std::is_same<TypeParam, rclcpp::Node>::value) {
+    component = std::make_shared<ComponentInterfacePublicInterface<TypeParam>>(
+        node_options, modulo_core::communication::PublisherType::PUBLISHER);
+  } else if (std::is_same<TypeParam, rclcpp_lifecycle::LifecycleNode>::value) {
+    component = std::make_shared<ComponentInterfacePublicInterface<TypeParam>>(
+        node_options, modulo_core::communication::PublisherType::LIFECYCLE_PUBLISHER);
+  }
```

Comment:
nitpick: this is duplicated, can you just make it a simple templated function and call it instead?

```c++
template <typename T>
std::shared_ptr<ComponentInterfacePublicInterface<T>> make_component_from_options(const rclcpp::NodeOptions& options) {
  if (std::is_same<T, rclcpp::Node>::value) {
     return std::make_shared<ComponentInterfacePublicInterface<T>>(
         node_options, modulo_core::communication::PublisherType::PUBLISHER);
   } else if (std::is_same<T, rclcpp_lifecycle::LifecycleNode>::value) {
     return = std::make_shared<ComponentInterfacePublicInterface<T>>(
         node_options, modulo_core::communication::PublisherType::LIFECYCLE_PUBLISHER);
   }
}
```